hadoop-hdfs-project/hadoop-hdfs-rbf/src/main/java/org/apache/hadoop/hdfs/server/federation/router/ThreadLocalContext.java

```diff
@@ -75,6 +75,7 @@ public ThreadLocalContext() {
    * that the task execution reflects the state of the original calling thread.
    */
   public void transfer() {
+    Server.getCurCall().set(null);
```

Comment:
@hfutatzhanghb Thanks for the contribution! Why do we set it to null?

Reply:
@slfan1989 Thanks a lot for your reviewing, we should reset `Server.getCurCall` here to avoid async responder thread use wrong ThreadLocal variable.  For example, router always receives `safemode_get` response from NameNode, if we don't reset `Server.getCurCall()`, we will find the call information occurs in the debug log which is not accurate.

Reply:
Here is a screenshot which can better explain this case.
![image](https://github.com/user-attachments/assets/33f2d6f0-c11a-4ff6-8fc4-1bfc97e033cd)